File: Flutter-App/app/lib/api/models/user_invitation.dart
class UserInvitation {
  String? id;
  String? registrationToken;
  String? invitedBy;
  String? email;
  List<String>? readAccess;
  List<String>? writeAccess;
  bool? isTenantAdmin;
  bool? isAccepted;
  DateTime? expirationDate;
  DateTime? acceptedDate;
  bool? isDeleted;
  String? tenantId;

  UserInvitation(
      {this.id,
      this.registrationToken,
      this.invitedBy,
      this.email,
      this.readAccess,
      this.writeAccess,
      this.isTenantAdmin,
      this.isAccepted,
      this.expirationDate,
      this.acceptedDate,
      this.isDeleted,
      this.tenantId});

  UserInvitation.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    registrationToken = json['registrationToken'];
    invitedBy = json['invitedBy'];
    email = json['email'];
    readAccess = json['readAccess'].cast<String>();
    writeAccess = json['writeAccess'].cast<String>();
    isAccepted = json['isAccepted'];
    expirationDate =DateTime.parse(json['expirationDate'].toString());
    isDeleted = json['isDeleted'];
    tenantId = json['tenantId'];
    isTenantAdmin = json['isTenantAdmin'];
  }

  Map<String, dynamic> toCreateJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['email'] = email;
    data['tenantId'] = tenantId;
    data['readAccess'] = readAccess;
    data['writeAccess'] = writeAccess;
    data['isTenantAdmin'] = isTenantAdmin;
    return data;
  }
}

File: Backend/Core/Entities/UserInvitation.cs
namespace Core.Entities;

public class UserInvitation
{
    public Guid Id { get; set; }
    public Guid RegistrationToken { get; set; }
    public string InvitedBy { get; set; } = null!;
    public string Email { get; set; } = null!;
    public bool IsTenantAdmin { get; set; }
    public List<string> ReadAccess { get; set; } = new();
    public List<string> WriteAccess { get; set; } = new();
    public bool IsAccepted { get; set; }
    public DateTime ExpirationDate { get; set; }
    public DateTime? AcceptedDate { get; set; }
    public bool IsDeleted { get; set; }
    public Guid TenantId { get; set; }
}